### DISCUSSION TITLE
Upgrade panda version break my recipe JSX tracking

### DISCUSSION THREAD

**Discussion by azoriusss** [chakra-ui/panda#2210] [Q&A] [answered]
Hi
I'm facing two strange issues when upgrading panda

**First issue : I can't upgrade** 

First of all, running `yarn upgrade "@pandacss/*"` is not updating anything in my `package.json` file. 

I need to manually change version in the file and run `yarn` 

my yarn version is `1.22.19` and I'm not an expert with yarn (so I'm possibly making something stupid here)

**Second issue** 
When updating above `0.21`, JSX usage of my recipes (any of them) is not tracked anymore. The only remaining generated styles are coming from recipe `base`, or from `defaultVariant`

First, I thought it was linked to my way of updating panda, but today I tried to reproduce my usage in the playground and I am facing the same issue 
[See playground here](https://play.panda-css.com/eZq-AMPCOj)

I tried to add the name of my compoents (even if they are the same as the recipe) in `jsx` option of my recipe, but it doesn't change anything

Did I miss something through updates ? Are this 2 issues linked in any way ?

Thanks in advance for your help :pray:

**Comment by segunadebayo** [comment, marked as answer]
Hi @vlrd-playin,

In a few versions back, we stopped tracking JSX elements without an explicit `jsxFramework` set in the config.

Add `jsxFramework: "react" to your config to fix this.

See the diff here: https://play.panda-css.com/eZq-AMPCOj/Rp2aTUqUv2

**Reply by azoriusss** [reply]
Thank you for this quick answer
It works as expected now ! 

FYI it's not specified in [this section from the doc](https://panda-css.com/docs/concepts/recipes#usage-in-jsx-1) (for recipes). The warning is only present in the cva section ! I should have tried it anyway but it can be confusing
